### DISCUSSION TITLE
NXP i.MX RT595 Evaluation Kit QSPI driver display support

### DISCUSSION THREAD

**Discussion by jdwang001** [zephyrproject-rtos/zephyr#50175] [General] [answered]
I want to use rt595 to drive the display through qspi, but I only see flexcomm configuration spi related content in dts。
How can I configure flexcomm as qspi in dts?

**Comment by DerekSnell** [comment, marked as answer]
Hi @jdwang001 ,

You have a display that is controlled through a 4-bit SPI interface, like a QSPI flash device?  That is the 1st I have heard of a display like that

Let’s review the RT500 peripherals related to your question:
- FlexSPI – A high-speed interface intended for external memories with quad or octal interfaces.  The RT500 has two FlexSPI peripherals, and can interface to many memories including QSPI.
- Flexcomm – flexible serial communication interface, can be configured as 1-bit SPI, I2C, UART or I2S.  RT500 has many Flexcomm peripherals available.
- HS SPI – the High-Speed SPI peripherals are Flexcomms that have been stripped down to support only 1-bit SPI interface for higher frequencies.  RT500 has 2 HS_SPI peripherals
- Other display interfaces – the RT500 peripherals intended to interface with a display include parallel interfaces from LCDIF or FlexIO, or serial with MIPI DSI.

So to answer your questions, you cannot configure a Flexcomm for 4-bit QSPI interface.  It will only support a traditional 1-bit SPI.  However, if the display is similar to QSPI memory devices, then potentially you could use a FlexSPI as the interface.  I am not aware of any example having FlexSPI drive a display.  But there are many examples using the FlexSPI to interface to different memory devices, including QSPI flash.  The RT500 EVK has the 2 FlexSPIs connected to an octal DDR flash and an octal DDR pSRAM.  The EVK also has a QSPI flash on the board that can connect to FlexSPI0.  But this requires hardware modifications to the board, and has not yet been enabled in Zephyr.

Another board that could be a good reference for the QSPI interface in Zephyr is the MIMXRT1060-EVKB board.  That board defaults to a QSPI flash on the FlexSPI.  It is the same IS25WP064 QSPI flash placed on the RT500 EVK, and the Zephyr board support already has this QSPI enabled.  So you can use the MIMXRT1060-EVKB board as a reference for the devicetree and FlexSPI driver configuration used in Zephyr.

To use the FlexSPI with a display, I don’t know if you will be able to use the Zephyr QSPI FlexSPI driver as is, or if you will need to customize the driver for your display.  The driver used in Zephyr for the QSPI devices is [flash_mcux_flexspi_nor.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/flash_mcux_flexspi_nor.c).  If you need to tweak this driver for the display, you can read the FlexSPI chapter in the [RT500 Reference Manual](https://www.nxp.com/webapp/Download?colCode=IMXRT500RM) for all the details on the FlexSPI and how to configure it.

Best regards

**Reply by jdwang001** [reply]
Wow, I am very happy to receive a reply.
Yes, I mistook flexcomm for supporting qspi.
I intend to try to modify the QSPI driver on RT500 to support my current QSPI display.
THX， @DerekSnell 。
